apps/api/src/server/applications/sector/__tests__/get-sectors.test.js

```diff
@@ -89,3 +91,87 @@ describe('Get sectors', () => {
 		});
 	});
 });
+
+test('test if sector cache is working', async () => {
+	nodeCache.flushAll();
+
+	const cacheSector = {
+		id: 3,
+		name: 'cache test name',
+		abbreviation: 'CC',
+		displayNameEn: 'cache test name en',
+		displayNameCy: 'cache test name cy'
+	};
+
+	setCache('', [cacheSector]);
+
+	const response = await request.get('/applications/sector');
+
+	expect(response.status).toEqual(200);
+	expect(response.body).toEqual([
+		{
+			name: cacheSector.name,
+			abbreviation: cacheSector.abbreviation,
+			displayNameEn: cacheSector.displayNameEn,
+			displayNameCy: cacheSector.displayNameCy
+		}
+	]);
+});
+
+const sectorName = 'Test1';
+
+test('test if cache is working with specific string', async () => {
+	nodeCache.flushAll();
+
+	const cacheSector = {
+		id: 3,
+		name: 'cache test name',
+		abbreviation: 'CC',
+		displayNameEn: 'cache test name en',
+		displayNameCy: 'cache test name cy'
+	};
+
+	setCache('Test1', [cacheSector]);
+
+	databaseConnector.sector.findUnique.mockResolvedValue([sector]);
+
+	const response = await request.get('/applications/sector').query({ sectorName });
+
+	expect(response.status).toEqual(200);
+	expect(response.body).toEqual([
+		{
+			name: cacheSector.name,
+			abbreviation: cacheSector.abbreviation,
+			displayNameEn: cacheSector.displayNameEn,
+			displayNameCy: cacheSector.displayNameCy
+		}
+	]);
+});
+
+test('test if cache is working with specific string', async () => {
+	nodeCache.flushAll();
+
+	const cacheSector = {
+		id: 3,
+		name: 'cache test name',
+		abbreviation: 'CC',
+		displayNameEn: 'cache test name en',
+		displayNameCy: 'cache test name cy'
+	};
+
+	setCache('Test1', [cacheSector]);
+
+	databaseConnector.sector.findUnique.mockResolvedValue([sector]);
+
+	const response = await request.get('/applications/sector').query({ sectorName });
+
+	expect(response.status).toEqual(200);
+	expect(response.body).toEqual([
+		{
+			name: cacheSector.name,
+			abbreviation: cacheSector.abbreviation,
+			displayNameEn: cacheSector.displayNameEn,
+			displayNameCy: cacheSector.displayNameCy
+		}
+	]);
```

Comment:
I would say that here to double check our word, we should check that the `databaseConnector.sector.findUnique` function has definitely not been called

Reply:
'double check our word' - what do you mean? It is returning the cacheSector object in this test - if you mean to remove the findUnique fx, if I do that, the test does not pass.

Reply:
lol i have no idea what that sentence was mean to be :D 

Yes, i can see that the returned data is as we would expect. How come the tests don't pass?

Reply:
lol i have no idea what that sentence was mean to be :D 

Yes, i can see that the returned data is as we would expect. How come the tests don't pass?

Reply:
the tests are passing with the above - it returns 400 if I remove `databaseConnector.sector.findUnique.mockResolvedValue([sector]);` as I guess this is the call to the db

Reply:
Yeah looks like a bug to me

Reply:
so, do you mean to find a way to resolve this that does not use (i.e. remove) this line: `databaseConnector.sector.findUnique.mockResolvedValue([sector]);` ?

Reply:
Essentially I think yes. Basically we want to test that we don't make any calls to the database at all